cargo-nextest/src/dispatch.rs

```diff
@@ -316,38 +414,71 @@ impl AppImpl {
     fn exec(self) -> Result<()> {
         let output = self.output.init();
 
-        let graph = build_graph(self.manifest_path.as_deref(), output)?;
-
         match self.command {
             Command::List {
                 build_filter,
                 message_format,
+                no_query,
+                output_binaries_metadata,
+                output_cargo_metadata,
+                ref no_build,
             } => {
+                let graph = build_graph(
+                    self.manifest_path.as_deref(),
+                    output,
+                    no_build,
+                    output_cargo_metadata.as_deref(),
+                )?;
+
                 let target_runner = runner_for_target(build_filter.cargo_options.target.as_deref());
 
-                let mut test_list = build_filter.compute(
-                    self.manifest_path.as_deref(),
+                let binary_list = build_binary_list(
                     &graph,
+                    &build_filter,
+                    self.manifest_path.as_deref(),
                     output,
-                    target_runner.as_ref(),
+                    no_build,
+                    output_binaries_metadata.as_deref(),
                 )?;
-                if output.color.should_colorize(Stream::Stdout) {
-                    test_list.colorize();
+
+                if no_query {
+                    write_to_stdout(|writer| {
+                        binary_list
+                            .write(
+                                message_format.to_output_format(output.verbose),
+                                writer,
+                                output.color.should_colorize(Stream::Stdout),
+                            )
+                            .map_err(Into::into)
+                    })?;
```

Comment:
It's not entirely clear to me what's happening here -- it seems like the binaries metadata might get written out twice, once to stdout and once to what's specified in `output-binaries-metadata`? This seems somewhat incorrect to me. I think `output_binaries_metadata` should just not exist.

Also, why add `output_cargo_metadata` when users can just run `cargo metadata` anyway? Some scripting around this is going to be required in any case -- seems simpler to just have consumers run `cargo metadata --format-version 1` themselves.

Reply:
Indeed.

I did get lost / trapped in my design reflection without your previous comment about `--include-cargo-metadata` and wasn't sure about exposing the use to `cargo metadata` directly (they must use the right arguments).

But indeed it makes more sense to let them call it.

I removed both `--output-*` parameters, and will work on a proper documentation for the feature.